### PR TITLE
Added 2D Shepp Logan Demo

### PR DESCRIPTION
**Pull request by cabouman**

### CODE REVIEW

File: demo/demo_2D_shepp_logan.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+import svmbir
+
+num_rows = 256
+num_cols = num_rows
+num_views = 144
+threshold = 0.05
+
```
Comment (gbuzzard) [inline demo/demo_2D_shepp_logan.py:10]: I would include an overview of this file.  Something like
"""
This file demonstrates the generation of a Shepp-Logan phantom followed by sinogram projection and reconstruction using MBIR. 
"""

File: demo/demo_2D_shepp_logan.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+print(f'The RMSE reconstruction error is {rmse:.3f}.')
+
+## display phantom
+imgplot = plt.imshow(phantom[0],vmin=1.0,vmax=1.1)
+imgplot.set_cmap('gray')
+plt.colorbar()
+plt.savefig('output/shepp_logan_phantom.png')
+plt.close()
+
+## display sharper reconstruction
```
Comment (gbuzzard) [inline demo/demo_2D_shepp_logan.py:37]: Remover the word "sharper" from line 37

File: demo/demo_2D_shepp_logan.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+rmse = np.sqrt(np.linalg.norm(recon[0] - phantom[0]) ** 2 / (recon.shape[1] * phantom.shape[2]))
+print(f'The RMSE reconstruction error is {rmse:.3f}.')
+
+## display phantom
+imgplot = plt.imshow(phantom[0],vmin=1.0,vmax=1.1)
+imgplot.set_cmap('gray')
+plt.colorbar()
+plt.savefig('output/shepp_logan_phantom.png')
+plt.close()
+
```
Comment (gbuzzard) [inline demo/demo_2D_shepp_logan.py:36]: Also display the sinogram.

File: demo/demo_2D_shepp_logan.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+## display sharper reconstruction
+imgplot = plt.imshow(recon[0],vmin=1.0,vmax=1.1)
+imgplot.set_cmap('gray')
+plt.colorbar()
+plt.savefig('output/shepp_logan_recon_sharp.png')
```
Comment (DamonLee5) [inline demo/demo_2D_shepp_logan.py]: sharpness=1.0 just uses the default value. Maybe we can change 'output/shepp_logan_recon_sharp.png' to 'output/shepp_logan_recon.png'

**Pull request closed without merging**